src/interactivity/win32/ut_interactivity_win32/UiaTextRangeTests.cpp

```diff
@@ -1346,4 +1361,312 @@ class UiaTextRangeTests
+
+            // TODO CARLOS: Fix this test when we actually know how to convert the font size into points
+            VERIFY_IS_TRUE(false);
+        }
+        {
+            Log::Comment(L"Test Read Only (special)");
+
+            VARIANT var{};
+            var.vt = VT_BOOL;
+            var.boolVal = false;
+
+            Microsoft::WRL::ComPtr<ITextRangeProvider> result;
+            VERIFY_SUCCEEDED(utr->FindAttribute(IsReadOnly, var, false, result.GetAddressOf()));
+
+            // Expecting the same text range endpoints
+            BOOL isEqual;
+            THROW_IF_FAILED(utr->Compare(result.Get(), &isEqual));
+            VERIFY_IS_TRUE(isEqual);
+
+            // Now perform the same test, but searching backwards
+            Log::Comment(L"Test Read Only (special) - Backwards");
+            Microsoft::WRL::ComPtr<ITextRangeProvider> resultBackwards;
+            VERIFY_SUCCEEDED(utr->FindAttribute(IsReadOnly, var, true, resultBackwards.GetAddressOf()));
+
+            // Expecting the same text range endpoints
+            THROW_IF_FAILED(result->Compare(resultBackwards.Get(), &isEqual));
+            VERIFY_IS_TRUE(isEqual);
+        }
+        {
+            Log::Comment(L"Test IsItalic (standard attribute)");
+
+            // Since all of the other attributes operate very similarly,
+            //  we're just going to pick one of them and test that.
+            // The "GetAttribute" tests provide code coverage for
+            //  retrieving an attribute verification function.
+            // This test is intended to provide code coverage for
+            //  finding a text range with the desired attribute.
+
+            // Set up the buffer's attributes.
+            TextAttribute italicAttr;
+            italicAttr.SetItalic(true);
+            auto iter{ _pUiaData->GetTextBuffer().GetCellDataAt(startPos) };
+            for (auto i = 0; i < 5; ++i)
+            {
+                _pTextBuffer->Write({ italicAttr }, iter.Pos());
+                ++iter;
+            }
+
+            // set the expected end (exclusive)
+            auto expectedEndPos{ iter.Pos() };
+            _pUiaData->GetTextBuffer().GetSize().IncrementInBounds(expectedEndPos);
+
+            VARIANT var{};
+            var.vt = VT_BOOL;
+            var.boolVal = true;
+
+            Microsoft::WRL::ComPtr<ITextRangeProvider> result;
+            VERIFY_SUCCEEDED(utr->FindAttribute(Italic, var, false, result.GetAddressOf()));
+
+            // TODO CARLOS: The "end" verification is failing. Here's a trace of what's going on:
+            // - UiaTextRangeBase:
+            //   - checkIfAttrFound is properly set and used
+            //   - the search loop properly sets the "start"
+            //   - the search loop iterates through the entire utr because checkIfAttrFound
+            //      continues to return true for the whole range
+            // I have no clue why this is happening. A fresh pair of eyes would be appreciated here :)

```

Comment:
I have no clue why this is failing. For any reviewer that comes along, please provide some ideas on what to look at here. Is my `checkIfAttrFound` written improperly?

Reply:
For anybody that was curious, I messed up by looking at `utr` instead of `result`. So I was straight up comparing the wrong UTR hahaha.